### PR TITLE
More base luthier factories

### PR DESCRIPTION
**Pull request by michael-mclawhorn**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/DoubleMaxMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/DoubleMinMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/logicalmetric/DefaultMetricFactory.java  7
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/keyvaluestore/MapKeyValueStoreFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/searchprovider/NoOpSearchProviderFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/searchprovider/ScanSearchProviderFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/logicaltable/DefaultLogicalTableGroupFactory.java  3
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongMinMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongSumMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongMaxMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/DoubleSumMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/LuthierPhysicalTableParams.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/ArithmeticMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/CountMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/searchprovider/LuceneSearchProviderFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/PermissivePhysicalTableFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/dimension/KeyValueStoreDimensionFactory.java  6
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/StrictPhysicalTableFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/AggregationAverageMakerFactory.java  3
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/SingleDataSourcePhysicalTableFactory.java  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fili/fili/pullRequest?prid=4529265&bid=15247446)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/DoubleMinMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/searchprovider/NoOpSearchProviderFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/LuthierPhysicalTableParams.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/keyvaluestore/MapKeyValueStoreFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/DoubleMaxMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongSumMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/CountMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/logicaltable/DefaultLogicalTableGroupFactory.java  3
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/searchprovider/LuceneSearchProviderFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/PermissivePhysicalTableFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/dimension/KeyValueStoreDimensionFactory.java  6
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/SingleDataSourcePhysicalTableFactory.java  9
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/AggregationAverageMakerFactory.java  3
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongMinMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/searchprovider/ScanSearchProviderFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/ArithmeticMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/StrictPhysicalTableFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/DoubleSumMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/logicalmetric/DefaultMetricFactory.java  7
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongMaxMakerFactory.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fili/fili/pullRequest?prid=4529265&bid=15247446)

File: luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongMinMakerFactory.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import com.yahoo.bard.webservice.data.config.metric.makers.LongMaxMaker;
+import com.yahoo.bard.webservice.data.config.metric.makers.MetricMaker;
+
+/**
+ * Factory that supports LongMinMaker.
+ */
+public class LongMinMakerFactory implements Factory<MetricMaker> {
+    @Override
+    public MetricMaker build(String name, LuthierConfigNode configTable, LuthierIndustrialPark resourceFactories) {
+        return new LongMaxMaker(resourceFactories.getMetricDictionary());
```
Comment (archolewa) [inline luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongMinMakerFactory.java]: Typo. Pretty sure you mean LongMinMaker here.

**Comment by archolewa** [comment]
Just the one comment. The rest look good. :+1: once that's been addressed.

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/dimension/KeyValueStoreDimensionFactory.java  6
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/searchprovider/NoOpSearchProviderFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/PermissivePhysicalTableFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/CountMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/DoubleMinMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/SingleDataSourcePhysicalTableFactory.java  9
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/logicalmetric/DefaultMetricFactory.java  7
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/DoubleMaxMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/logicaltable/DefaultLogicalTableGroupFactory.java  3
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/ArithmeticMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/searchprovider/LuceneSearchProviderFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/AggregationAverageMakerFactory.java  3
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/LuthierPhysicalTableParams.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongSumMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongMaxMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/StrictPhysicalTableFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/DoubleSumMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/searchprovider/ScanSearchProviderFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongMinMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/keyvaluestore/MapKeyValueStoreFactory.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fili/fili/pullRequest?prid=4529265&bid=15247446)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/LuthierPhysicalTableParams.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/SingleDataSourcePhysicalTableFactory.java  9
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/StrictPhysicalTableFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/DoubleMinMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/physicaltable/PermissivePhysicalTableFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/ArithmeticMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/logicaltable/DefaultLogicalTableGroupFactory.java  3
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/DoubleSumMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/keyvaluestore/MapKeyValueStoreFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/logicalmetric/DefaultMetricFactory.java  7
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongMinMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/searchprovider/ScanSearchProviderFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/searchprovider/LuceneSearchProviderFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/dimension/KeyValueStoreDimensionFactory.java  6
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/DoubleMaxMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/CountMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongSumMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/LongMaxMakerFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/searchprovider/NoOpSearchProviderFactory.java  1
- luthier/src/main/java/com/yahoo/bard/webservice/data/config/luthier/factories/metricmaker/AggregationAverageMakerFactory.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fili/fili/pullRequest?prid=4529265&bid=15247446)

**Pull request merged**